client.go

```diff
@@ -360,4 +482,100 @@ func (c *Client) createComponents(config *Config) {
 		Logger:               c.ic.Logger,
 	}
 	c.proxyManager = newProxyManager(proxyManagerServiceBundle)
+	c.db = sql.OpenDB(idriver.NewConnectorWithClient(c.ic, true))
+}
+
+// SQLOptions are server-side query options.
+type SQLOptions struct {
+	cursorBufferSize int32
+	timeout          *int64
+	schema           *string
+	err              error
+}
+
+/*
+SetCursorBufferSize sets the query cursor buffer size.
+When rows are ready to be consumed, they are put into an internal buffer of the cursor.
+This parameter defines the maximum number of rows in that buffer.
+When the threshold is reached, the backpressure mechanism will slow down the execution, possibly to a complete halt, to prevent out-of-memory.
+The default value is expected to work well for most workloads.
+A bigger buffer size may give you a slight performance boost for queries with large result sets at the cost of increased memory consumption.
+Defaults to 4096.
+The given buffer size must be in the non-negative int32 range.
+*/
+func (s *SQLOptions) SetCursorBufferSize(cbs int) {
+	v, err := check.NonNegativeInt32(cbs)
+	if err != nil {
+		s.err = ihzerrors.NewIllegalArgumentError("setting cursor buffer size", err)
+		return
+	}
+	s.cursorBufferSize = v
+}
+
+/*
+SetQueryTimeout sets the query execution timeout.
+If the timeout is reached for a running statement, it will be cancelled forcefully.
+Zero value means no timeout.
+Negative values mean that the value from the server-side config will be used.
+Defaults to -1.
+*/
+func (s *SQLOptions) SetQueryTimeout(t time.Duration) {
+	if t == 0 {
+		s.timeout = nil
+		return
+	}
+	tm := t.Milliseconds()
+	// note that the condition below is for t, not tm
+	if t < 0 {
+		tm = -1
+	}
+	s.timeout = &tm
+}
+
+/*
+SetSchema sets the schema name.
+The engine will try to resolve the non-qualified object identifiers from the statement in the given schema.
+If not found, the default search path will be used.
```

Comment:
```suggestion
If not found, the default search path will be used, which looks for objects in the predefined schemas "partitioned" and "public".
```

We may add this part which exist in Java client apidoc.

Reply:
Updated at: https://github.com/hazelcast/hazelcast-go-client/pull/728/commits/6d89765f1e4b6dceb68f93e76191d3facb572ae9